agent/lib/kontena/observer.rb

```diff
@@ -1,117 +1,346 @@
+module Celluloid
+  # XXX: allow Celluloid.exclusive { ... } to be used outside of actor context: it's a no-op
+  def exclusive(&block)
```

Comment:
Shouldn't be monkey-patching `Celluloid` here in this file, either:

* Move this to `lib/celluloid/exclusive_patch.rb`
* Wrap this in `Kontena::Observer.exclusive`